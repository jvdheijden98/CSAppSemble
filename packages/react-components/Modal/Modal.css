.enter {
  opacity: 0;
}

.enterActive {
  opacity: 1;
  transition: opacity 90ms ease-in;
}

.exit {
  opacity: 1;
}

.exitActive {
  opacity: 0;
  transition: opacity 90ms ease-in;
}

.modal {
  z-index: 3000 !important;
}

.closeButton {
  position: absolute;
  right: 10px;
  top: 10px;
  z-index: 1;
}
