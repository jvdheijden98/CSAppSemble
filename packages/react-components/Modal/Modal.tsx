import classNames from 'classnames';
import * as React from 'react';
import { WrappedComponentProps } from 'react-intl';
import { CSSTransition } from 'react-transition-group';

import messages from './messages';
import styles from './Modal.css';

export interface ModalProps extends WrappedComponentProps {
  /**
   * The child elements to render on the modal.
   */
  children?: React.ReactNode;
  /**
   * Wether or not the modal is currently active.
   */
  isActive: boolean;
  /**
   * A function that will be called when the user closes the modal.
   */
  onClose?: React.ReactEventHandler;

  /**
   * The title that is displayed at the top of the modal.
   */
  title?: React.ReactNode;

  /**
   * The CSS class applied to the body
   */
  className?: string;
}

/**
 * Render an aria compliant modal overlay.
 */
export default class Modal extends React.Component<ModalProps> {
  static defaultProps: Partial<ModalProps> = {
    children: null,
    onClose() {},
  };

  onKeyDown: React.KeyboardEventHandler = event => {
    const { onClose } = this.props;

    if (event.key === 'Escape') {
      onClose(event);
    }
  };

  render(): JSX.Element {
    const { children, className, intl, isActive, onClose, title } = this.props;

    return (
      <CSSTransition
        classNames={{
          enter: styles.enter,
          enterActive: styles.enterActive,
          exit: styles.exit,
          exitActive: styles.exitActive,
        }}
        in={isActive}
        mountOnEnter
        timeout={90}
        unmountOnExit
      >
        <div className={`is-active modal ${styles.modal}`}>
          <div
            className="modal-background"
            onClick={onClose}
            onKeyDown={this.onKeyDown}
            role="presentation"
          />
          <div className="modal-card">
            <div className="modal-card-head">
              <p className="modal-card-title">{title}</p>
              <button
                aria-label={intl.formatMessage(messages.closeDialog)}
                className="delete is-large"
                onClick={onClose}
                type="button"
              />
            </div>
            <div className={classNames('modal-card-body', className)}>{children}</div>
          </div>
        </div>
      </CSSTransition>
    );
  }
}
