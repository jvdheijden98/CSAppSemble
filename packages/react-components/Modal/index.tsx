import { injectIntl, WrappedComponentProps } from 'react-intl';

import Modal, { ModalProps } from './Modal';

export default injectIntl<'intl', WrappedComponentProps & ModalProps>(Modal);
