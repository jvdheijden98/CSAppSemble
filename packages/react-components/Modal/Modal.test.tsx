import { shallow } from 'enzyme';
import React from 'react';
import { createIntl, IntlShape } from 'react-intl';

import Modal from './Modal';

let intl: IntlShape;

beforeEach(() => {
  intl = createIntl({ locale: 'en' });
});

it('should not render a bulma modal when it is inactive', () => {
  const wrapper = shallow(
    <Modal intl={intl} isActive={false} onClose={() => {}}>
      test
    </Modal>,
  );
  expect(wrapper).toMatchSnapshot();
});

it('should render a bulma modal when it is active', () => {
  const wrapper = shallow(
    <Modal intl={intl} isActive onClose={() => {}}>
      test
    </Modal>,
  );
  expect(wrapper).toMatchSnapshot();
});

it('should close the modal when the close button is clicked', () => {
  const onClose = jest.fn();
  const wrapper = shallow(
    <Modal intl={intl} isActive onClose={onClose}>
      test
    </Modal>,
  );
  wrapper.find('.delete').simulate('click');
  expect(onClose).toHaveBeenCalled();
});

it('should close the modal when the background is clicked', () => {
  const onClose = jest.fn();
  const wrapper = shallow(
    <Modal intl={intl} isActive onClose={onClose}>
      test
    </Modal>,
  );
  wrapper.find('.modal-background').simulate('click');
  expect(onClose).toHaveBeenCalled();
});

it('should close the modal escape is pressed on the background', () => {
  const onClose = jest.fn();
  const wrapper = shallow(
    <Modal intl={intl} isActive onClose={onClose}>
      test
    </Modal>,
  );
  wrapper.find('.modal-background').simulate('keydown', { key: 'Escape' });
  expect(onClose).toHaveBeenCalled();
});

it('should not close the modal another key is pressed on the background', () => {
  const onClose = jest.fn();
  const wrapper = shallow(
    <Modal intl={intl} isActive onClose={onClose}>
      test
    </Modal>,
  );
  wrapper.find('.modal-background').simulate('keydown', { key: 'Space' });
  expect(onClose).not.toHaveBeenCalled();
});
