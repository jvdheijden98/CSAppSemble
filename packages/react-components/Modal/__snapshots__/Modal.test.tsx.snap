// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render a bulma modal when it is inactive 1`] = `
<CSSTransition
  classNames={
    Object {
      "enter": "enter",
      "enterActive": "enterActive",
      "exit": "exit",
      "exitActive": "exitActive",
    }
  }
  in={false}
  mountOnEnter={true}
  timeout={90}
  unmountOnExit={true}
>
  <div
    className="is-active modal modal"
  >
    <div
      className="modal-background"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
    <div
      className="modal-card"
    >
      <div
        className="modal-card-head"
      >
        <p
          className="modal-card-title"
        />
        <button
          aria-label="Close dialog"
          className="delete is-large"
          onClick={[Function]}
          type="button"
        />
      </div>
      <div
        className="modal-card-body"
      >
        test
      </div>
    </div>
  </div>
</CSSTransition>
`;

exports[`should render a bulma modal when it is active 1`] = `
<CSSTransition
  classNames={
    Object {
      "enter": "enter",
      "enterActive": "enterActive",
      "exit": "exit",
      "exitActive": "exitActive",
    }
  }
  in={true}
  mountOnEnter={true}
  timeout={90}
  unmountOnExit={true}
>
  <div
    className="is-active modal modal"
  >
    <div
      className="modal-background"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    />
    <div
      className="modal-card"
    >
      <div
        className="modal-card-head"
      >
        <p
          className="modal-card-title"
        />
        <button
          aria-label="Close dialog"
          className="delete is-large"
          onClick={[Function]}
          type="button"
        />
      </div>
      <div
        className="modal-card-body"
      >
        test
      </div>
    </div>
  </div>
</CSSTransition>
`;
